From: =?UTF-8?q?Erik=20Spr=C3=A5ng?= <sprang@webrtc.org>
Date: Wed, 23 Oct 2024 11:20:12 +0200
Subject: (cherry-pick-branch-heads/6778) Merge corruption detection fixes to
 M131.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This CL comprises the following CLs:

Do not change crypto options in peer_connection.cc

Bug: webrtc:358039777, chromium:374122158
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/365360
Commit-Queue: Emil Vardar (xWF) <vardar@google.com>
Reviewed-by: Henrik Boström <hbos@webrtc.org>
Reviewed-by: Erik Språng <sprang@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#43245}

Enable corruption detection when the encrypted extension is present

Credit: https://webrtc-review.googlesource.com/c/src/+/365584 with ASAN issue solved.

Bug: webrtc:358039777, chromium:374122158
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/365680
Commit-Queue: Emil Vardar (xWF) <vardar@google.com>
Reviewed-by: Erik Språng <sprang@webrtc.org>
Reviewed-by: Ilya Nikolaevskiy <ilnik@webrtc.org>
Cr-Commit-Position: refs/heads/main@{#43244}

Post corruption score aggregation to worker thread.

Bug: webrtc:358039777, chromium:374122158
Change-Id: Iff89c21657939557a10b6a183d632f251086379c
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/365600
Reviewed-by: Erik Språng <sprang@webrtc.org>
Commit-Queue: Åsa Persson <asapersson@webrtc.org>
Auto-Submit: Emil Vardar (xWF) <vardar@google.com>
Reviewed-by: Åsa Persson <asapersson@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#43238}
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/366521
Reviewed-by: Henrik Boström <hbos@webrtc.org>
Commit-Queue: Erik Språng <sprang@webrtc.org>
Cr-Commit-Position: refs/branch-heads/6778@{#1}
Cr-Branched-From: 7b1b7a0f51593df7a1a802f489d6a2fb14195bcc-refs/heads/main@{#43221}
---
 pc/peer_connection.cc | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/pc/peer_connection.cc b/pc/peer_connection.cc
index a6eda6f602..1171bc8929 100644
--- a/pc/peer_connection.cc
+++ b/pc/peer_connection.cc
@@ -761,11 +761,6 @@ RTCError PeerConnection::Initialize(
 
   configuration_ = configuration;
 
-  if (configuration_.crypto_options) {
-    configuration_.crypto_options->srtp.enable_encrypted_rtp_header_extensions =
-        trials().IsEnabled("WebRTC-EncryptedRtpHeaderExtensions");
-  }
-
   legacy_stats_ = std::make_unique<LegacyStatsCollector>(this);
   stats_collector_ = RTCStatsCollector::Create(this, env_);
 
@@ -820,8 +815,6 @@ JsepTransportController* PeerConnection::InitializeTransportController_n(
   config.crypto_options = configuration.crypto_options.has_value()
                               ? *configuration.crypto_options
                               : options_.crypto_options;
-  config.crypto_options.srtp.enable_encrypted_rtp_header_extensions =
-      trials().IsEnabled("WebRTC-EncryptedRtpHeaderExtensions");
   config.transport_observer = this;
   config.rtcp_handler = InitializeRtcpCallback();
   config.un_demuxable_packet_handler = InitializeUnDemuxablePacketHandler();
